src/spikeinterface/core/core_tools.py

```diff
@@ -257,7 +257,7 @@ def write_binary_recording(
     file_path_dict = {segment_index: file_path for segment_index, file_path in enumerate(file_path_list)}
     for segment_index, file_path in file_path_dict.items():
         num_frames = recording.get_num_frames(segment_index=segment_index)
-        data_size_bytes = dtype_size_bytes * num_frames * num_channels
+        data_size_bytes = np.int64(dtype_size_bytes) * num_frames * num_channels
```

Comment:
To avoid this type of overflow errors, the general solution is to have num_frames and num_channels to python int which has an infinite size. 

The proble comes with num_frames or num_channels is a numpy scalar.